x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/field_inputs.tsx

```diff
@@ -0,0 +1,232 @@
+      rawValuesLookup: new Set(rawValues),
+    };
+  }, [column]);
+
+  const { inputValue: localValues, handleInputChange } = useDebouncedValue<WrappedValue[]>({
+    onChange: onChangeWrapped,
+    value: wrappedValues,
+  });
+
+  const onFieldSelectChange = useCallback(
+    (choice, index = 0) => {
+      const fields = [...localValues];
+      const newFieldName = indexPattern.getFieldByName(choice.field)?.displayName;
+      if (newFieldName != null) {
+        fields[index] = { id: generateId(), value: newFieldName };
+
+        // update the layer state
+        handleInputChange(fields);
+      }
+    },
+    [localValues, indexPattern, handleInputChange]
+  );
+
+  // diminish attention to adding fields alternative
+  if (localValues.length === 1) {
+    const [{ value }] = localValues;
+    return (
+      <>
+        <FieldSelect
+          fieldIsInvalid={false}
+          currentIndexPattern={indexPattern}
+          existingFields={existingFields}
+          operationByField={operationSupportMatrix.operationByField}
+          selectedOperationType={column?.operationType}
+          selectedField={value}
+          onChoose={onFieldSelectChange}
+        />
+        <NewBucketButton
+          data-test-subj={`indexPattern-terms-add-field`}
+          onClick={() => {
+            handleInputChange([
+              ...localValues,
+              { id: generateId(), value: undefined, isNew: true },
+            ]);
+          }}
+          label={i18n.translate('xpack.lens.indexPattern.terms.addField', {
+            defaultMessage: 'Add field',
+          })}
+        />
+      </>
+    );
+  }
+  const disableActions = localValues.length === 2 && localValues.some(({ isNew }) => isNew);
+  const localValuesFilled = localValues.filter(({ isNew }) => !isNew);
+  return (
+    <>
+      <DragDropBuckets
+        onDragEnd={(updatedValues: WrappedValue[]) => {
+          handleInputChange(updatedValues);
+        }}
+        onDragStart={() => {}}
+        droppableId="TOP_TERMS_DROPPABLE_AREA"
+        items={localValues}
+      >
+        {localValues.map(({ id, value, isNew }, index) => {
+          // need to filter the available fields for multiple terms
+          // * a scripted field should be removed
+          // * if a field has been used, should it be removed? Probably yes?
+          // * if a scripted field was used in a singular term, should it be marked as invalid for multi-terms? Probably yes?
+          const filteredOperationByField = Object.keys(operationSupportMatrix.operationByField)
+            .filter(
+              (key) =>
+                (!rawValuesLookup.has(key) && !indexPattern.getFieldByName(key)?.scripted) ||
+                key === value
+            )
+            .reduce((memo, key) => {
+              memo[key] = operationSupportMatrix.operationByField[key];
+              return memo;
+            }, {} as OperationSupportMatrix['operationByField']);
```

Comment:
to avoid type assertion:
```suggestion
            .reduce<OperationSupportMatrix['operationByField']>((memo, key) => {
              memo[key] = operationSupportMatrix.operationByField[key];
              return memo;
            }, {});
```